src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Editor/EditorStateModel.cs

```diff
@@ -0,0 +1,302 @@
+        private IVsWindowFrame _windowFrame;
+        private IProjectFileModelWatcher _projectFileModelWatcher;
+        private EditProjectFileVsFrameEvents _frameEventsListener;
+        private ITextBufferStateListener _textBufferStateListener;
+
+        [ImportingConstructor]
+        public EditorStateModel(IProjectThreadingService threadingService,
+            UnconfiguredProject unconfiguredProject,
+            [Import(typeof(SVsServiceProvider))] IServiceProvider serviceProvider,
+            IVsShellUtilitiesHelper shellHelper,
+            IExportFactory<IProjectFileModelWatcher> projectFileModelWatcherFactory,
+            IExportFactory<ITextBufferStateListener> textBufferListenerFactory,
+            ITextBufferManager textBufferManager)
+        {
+            _threadingService = threadingService;
+            _unconfiguredProject = unconfiguredProject;
+            _serviceProvider = serviceProvider;
+            _shellHelper = shellHelper;
+            _projectFileWatcherFactory = projectFileModelWatcherFactory;
+            _textBufferListenerFactory = textBufferListenerFactory;
+            _textBufferManager = textBufferManager;
+        }
+
+        #region Initialization/Destruction
+
+        /// <summary>
+        /// Called by anything attempting to open a project file editor window, usually by a command. This will show the window frame, creating it if
+        /// has not already been created.
+        /// </summary>
+        public async Task OpenEditorAsync()
+        {
+            await _threadingService.SwitchToUIThread();
+            lock (_lock)
+            {
+                // If the editor is already open, just show it and return
+                if (_currentState != EditorState.NoEditor)
+                {
+                    // If we're initializing, _windowFrame might be null. In that case, when the initialization code
+                    // is done, it'll take care of showing the frame.
+                    _windowFrame?.Show();
+                    return;
+                }
+                _currentState = EditorState.Initializing;
+            }
+
+            // First, open the frame
+            var loadedProjectEditorGuid = Guid.Parse(LoadedProjectFileEditorFactory.EditorFactoryGuid);
+            _windowFrame = _shellHelper.OpenDocumentWithSpecificEditor(_serviceProvider, _unconfiguredProject.FullPath, loadedProjectEditorGuid, Guid.Empty);
+
+            // Set up the project file watcher, so changes to the project file are detected and the buffer is updated.
+            _projectFileModelWatcher = _projectFileWatcherFactory.CreateExport();
+            _projectFileModelWatcher.Initialize();
+
+            // Finally, show the frame and move to EditorClean in lockstep
+            lock (_lock)
+            {
+                _windowFrame.Show();
+                _currentState = EditorState.EditorClean;
+                _projectFileModelWatcher.Initialize();
```

Comment:
A mistake. I had started to factor this part out so I could have the editorfactory actually start the initialization step and ran into the IVsWindowPane issue. I must have missed this when I put it back.